### PR TITLE
Add dimensionality inference when converting to napari layer

### PR DESCRIPTION
**Pull request by quantumjot**
Relates to https://github.com/lowe-lab-ucl/napari-btrack/issues/52

This lets the napari reader try to infer whether the data are 2D or 3D, based on the data in the file. It was previously hard-coded into the reader:

https://github.com/quantumjot/BayesianTracker/blob/8139868c06ba9ac4754299494cb49a7bf3160d5a/btrack/napari/reader.py#L82

It might be an idea to store this as an HDF5 attribute at some point, but in lieu of that this will work.

Also added a quick test to check the inference works.

### CODE REVIEW

File: btrack/core.py [hunk 1]
```diff
@@ -667,4 +667,8 @@ def to_napari(
 
         ndim = self.configuration.volume.ndim if ndim is None else ndim
 
-        return utils.tracks_to_napari(self.tracks, ndim, replace_nan=replace_nan)
+        return utils.tracks_to_napari(self.tracks, ndim=ndim, replace_nan=replace_nan)
+
+
+if __name__ == "__main__":
```
Comment (p-j-smith) [inline btrack/core.py:673]: just wondering why you've added this?

File: tests/test_utils.py [hunk 2]
```diff
@@ -211,6 +211,20 @@ def test_tracks_to_napari_incorrect_ndim(ndim: int):
         data, properties, graph = utils.tracks_to_napari(tracks, ndim=ndim)
 
 
+@pytest.mark.parametrize("ndim", [2, 3])
+def test_tracks_to_napari_ndim_inference(ndim: int):
+    """Test inferring the correct dimensions from track data when using
+    `tracks_to_napari`."""
+
+    # make three fake tracks with properties
```
Comment (p-j-smith) [inline tests/test_utils.py:219]: is this comment correct? It looks like only 1 fake track is being created

**Review by p-j-smith — COMMENTED** [review]
looks good, just a couple of small questions

File: tests/_utils.py [hunk 3]
```diff
@@ -18,16 +18,21 @@
...
 ) -> tuple[btrack.btypes.PyTrackObject, dict[str, Any]]:
     """Create a test object."""
 
     rng = np.random.default_rng(seed=RANDOM_SEED)
-
+    z = (
+        rng.uniform(0.0, 1000.0)
+        if ndim == btrack.constants.Dimensionality.THREE
+        else 0.0
+    )
```
Comment (paddyroddy) [inline tests/_utils.py:30]: Why not just move this to the below dictionary? Like for `ID`

File: btrack/core.py (hunk 1 above)
Comment (quantumjot) [reply btrack/core.py:673]: ahh - I knew I missed something when resolving the conflicts!

File: tests/test_utils.py (hunk 2 above)
Comment (quantumjot) [reply tests/test_utils.py:219]: I think that's fine for this test, since it only needs one to test the dimensionality

Comment (p-j-smith) [reply tests/test_utils.py:219]: ah yep, a single track is fine but I think the comment:

```
# make three fake tracks with properties
```

should be changed to:

```
# make a fake track with properties
```

just to make it clear for future people looking at the test

**Review by p-j-smith — APPROVED** [review]

**Pull request merged**